src/main/java/it/gov/pagopa/payhub/auth/service/a2a/ValidateClientCredentialsService.java

```diff
@@ -8,12 +8,19 @@
 @Service
 @Slf4j
 public class ValidateClientCredentialsService {
+	private final ClientService clientService;
+
 	public static final String ALLOWED_GRANT_TYPE = "client_credentials";
 	public static final String ALLOWED_SCOPE = "openid";
 
-	public void validate(String scope, String clientSecret) {
+	public ValidateClientCredentialsService(ClientService clientService) {
+		this.clientService = clientService;
+	}
+
+	public void validate(String clientId, String scope, String clientSecret) {
 		validateProtocolConfiguration(scope);
 		validateClientSecret(clientSecret);
+		clientService.verifyCredentials(clientId, clientSecret);
```

Comment:
I would create a new Service, AuthorizeClientCredentialsRequest and move there the method instead of this Service.
It's not a validation operation, but an authorization op